lib/haxe/src/org/apache/thrift/transport/THttpClient.hx

```diff
@@ -94,10 +114,241 @@ class THttpClient extends TTransport {
+        var portString = url_regexp.matched(3);
+        var request = url_regexp.matched(4);
+        if( request == "" )
+            request = "/";
+        var port = if ( portString == null || portString == "" ) ssl ? 443 : 80 else Std.parseInt(portString.substr(1, portString.length - 1));
+
+        var options = {
+            "hostname":host,
+            "port":port,
+            "path":request,
+            "method":post ? "POST" : "GET",
+            "agent":false,
+            "withCredentials":#if (haxe_ver >= 3.3) me.withCredentials #else false #end
+        };
+
+        var headersCode = "";
+        for( h in headers )
+            headersCode += 'req.setHeader("${h.header}","${h.value}");';
+
+        // stringify data
+        var sendCode = "";
+        var data = binaryPostData;
+
+        if(data != null) {
+            var bufStr = haxe.Json.stringify(js.node.buffer.Buffer.hxFromBytes(data)).replace("'", "\\'");
+            sendCode = "var dataObj = JSON.parse('" + bufStr + "', "
+            + " function(key, value) {"
+                + "return value && value.type === 'Buffer'"
+                + "? new Buffer(value.data)"
+                + ": value;"
+            + "});";
+            sendCode += "req.write(dataObj);";
+            headersCode += 'req.setHeader("Content-Length",${data.length});';
+        } else {
+            headersCode += 'req.setHeader("Content-Length", 0);';
+        }
+
+        var responseEncoding = 'binary';
+ 
+        // The async request the other Node process executes
+        var execString = "var http = require('http'), https = require('https'), fs = require('fs');"
+          + "var doRequest = http" + (ssl ? "s" : "") + ".request;"
+          + "var options = " + haxe.Json.stringify(options) + ";"
+          + "var responseText = '';"
+          + "var req = doRequest(options, function(response) {"
+          + "response.setEncoding(" + (responseEncoding != null ? "'" + responseEncoding + "'" : responseEncoding) + ");"
+          + "response.on('data', function(chunk) {"
+          + "  responseText += chunk;"
+          + "});"
+          + "response.on('end', function() {"
+          + "if(responseText.length) {"
+          + "  responseText = JSON.stringify(new Buffer(responseText, " + (responseEncoding != null ? "'" + responseEncoding + "'" : responseEncoding) + "));"
+          + "}"
+          + "process.stdout.write(JSON.stringify({err: null, data: {statusCode: response.statusCode, headers: response.headers, text: responseText}}));"
+          + "});"
+          + "response.on('error', function(error) {"
+          + "process.stdout.write(JSON.stringify({err: error, errorMessage : error.message}));"
+          + "});"
+          + "}).on('error', function(error) {"
+          + "process.stdout.write(JSON.stringify({err: error, errorMessage : error.message}));"
+          + "});"
+          + headersCode
+          + sendCode 
+          + "req.end();";
```

Comment:
Did you think about possible code injection issues? How do we prevent that? Tested?